examples/audiopanning/main.go

```diff
@@ -0,0 +1,232 @@
+This is an example using
+stereo audio panning.
+Intro:   0[s] - %[2]d[s]
+Loop:    %[2]d[s] - %[3]d[s]
+Current: %0.2[4]f[s]
+Panning: %.2f`, ebiten.CurrentTPS(), introLengthInSecond, introLengthInSecond+loopLengthInSecond, float64(pos)/float64(time.Second), g.panning)
+	ebitenutil.DebugPrint(screen, msg)
+
+	// draw image to show where the sound is at related to the screen
+	op := &ebiten.DrawImageOptions{}
+	op.GeoM.Translate(g.xpos-float64(img.Bounds().Dx()/2), screenHeight/2)
+	screen.DrawImage(img, op)
+}
+
+func (g *Game) Layout(outsideWidth, outsideHeight int) (int, int) {
+	return screenWidth, screenHeight
+}
+
+func main() {
+	// Decode image from a byte slice instead of a file so that
+	// this example works in any working directory.
+	// If you want to use a file, there are some options:
+	// 1) Use os.Open and pass the file to the image decoder.
+	//    This is a very regular way, but doesn't work on browsers.
+	// 2) Use ebitenutil.OpenFile and pass the file to the image decoder.
+	//    This works even on browsers.
+	// 3) Use ebitenutil.NewImageFromFile to create an ebiten.Image directly from a file.
+	//    This also works on browsers.
+	rawimg, _, err := image.Decode(bytes.NewReader(images.Ebiten_png))
+	if err != nil {
+		log.Fatal(err)
+	}
+	img, _ = ebiten.NewImageFromImage(rawimg, ebiten.FilterDefault)
+
+	ebiten.SetWindowSize(screenWidth*2, screenHeight*2)
+	ebiten.SetWindowTitle("Audio Panning Loop (Ebiten Demo)")
+	g := &Game{}
+	if err := ebiten.RunGame(g); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// StereoPanStream is an audio buffer that changes the stereo channel's signal
+// based on the Panning.
+type StereoPanStream struct {
+	*bytes.Reader
+	pan float64 // -1: left; 0: center; 1: right
+}
+
+const sqrt2div2 = 0.7071067811865476 // math.sqrt(2)/2.0
+const rad45 = math.Pi / 4            // 45º
+
+func (s *StereoPanStream) Read(p []byte) (n int, err error) {
+	n, err = s.Reader.Read(p)
+	if err != nil {
+		return
+	}
+
+	angle := rad45 * s.pan
+	acos := math.Cos(angle)
+	asin := math.Sin(angle)
+	ls := sqrt2div2 * (acos - asin)
+	rs := sqrt2div2 * (acos + asin)
+	for i := 0; i < len(p); i += 4 {
+		lc := int16(float64(int16(p[i])|int16(p[i+1])<<8) * ls)
+		rc := int16(float64(int16(p[i+2])|int16(p[i+3])<<8) * rs)
+
+		p[i] = byte(lc)
+		p[i+1] = byte(lc >> 8)
+		p[i+2] = byte(rc)
+		p[i+3] = byte(rc >> 8)
+	}
+	return
+}
+
+func (s *StereoPanStream) Close() error {
+	return nil
+}
+
+func (s *StereoPanStream) SetPan(pan float64) {
+	s.pan = math.Min(math.Max(-1, pan), 1)
+}
+
+func (s *StereoPanStream) Pan() float64 {
+	return s.pan
+}
+
+// NewStereoPanStream returns a new StereoPanStream with a shared buffer src.
+// The src's format must be linear PCM (16bits little endian, 2 channel stereo)
+// without a header (e.g. RIFF header). The sample rate must be same as that
+// of the audio context.
+//
+// The src can be shared by multiple buffers.
+func NewStereoPanStream(src []byte) *StereoPanStream {
```

Comment:
This function is not used anywhere?

Reply:
I copied from my implementation, but it isn't used in the example. Removed